slack_sdk/web/async_client.py

```diff
@@ -2992,6 +3000,166 @@ async def files_upload(
             kwargs["content"] = content
             return await self.api_call("files.upload", data=kwargs)
 
+    async def files_upload_v2(
+        self,
+        *,
+        # for sending a single file
+        filename: Optional[str] = None,  # you can skip this only when sending along with content parameter
+        file: Optional[Union[str, bytes, IOBase]] = None,
```

Comment:
@seratch - is this type hint correct? I tried to pass in a `io.BytesIO` object here as the `file` as I thought `IOBase` covers that as well as a real file handle (e.g. from `open("filename", "rb")`) - but it failed with `TypeError: object of type '_io.BytesIO' has no len()` - len comes from `_to_v2_file_upload_item` I believe. So either the type hint is incorrect or the implementation of `_to_v2_file_upload_item` needs tweaking? You could probably just do something like this?

```py
def _to_v2_file_upload_item(upload_file: Dict[str, Any]) -> Dict[str, Optional[Any]]:
    file = upload_file.get("file")
    content = upload_file.get("content")
    data: Optional[bytes] = None
    if file is not None:
        if isinstance(file, str):  # filepath
            with open(file.encode("utf-8", "ignore"), "rb") as readable:
                data = readable.read()
        elif isinstance(io.BufferedIOBase, io.RawIOBase):
            data = file.read()
        else: # bytes
            data = file
    elif content is not None:
        if isinstance(content, str):  # filepath
            data = content.encode("utf-8")
        elif isinstance(content, bytes):
            data = content
        else:
            raise SlackRequestError("The given content must be either filepath as str or data as bytes")

    filename = upload_file.get("filename")
    if upload_file.get("filename") is None and isinstance(file, str):
        # use the local filename if filename is missing
        if upload_file.get("filename") is None:
            filename = file.split(os.path.sep)[-1]
        else:
            filename = "Uploaded file"

    title = upload_file.get("title", "Uploaded file")
    if data is None:
        raise SlackRequestError(f"File content not found for filename: {filename}, title: {title}")

    if title is None:
        title = filename  # to be consistent with files.upload API

    return {
        "filename": filename,
        "data": data,
        "length": len(data),
        "title": title,
        "alt_txt": upload_file.get("alt_txt"),
        "snippet_type": upload_file.get("snippet_type"),
    }
```

Reply:
@Alex-ley-scrub Thanks for flagging this! This should be improved. We will look into this later.